packer/registry/image/image.go

```diff
@@ -131,10 +123,6 @@ func WithProvider(name string) func(*Image) error {
 // used to override the ImageId for an existing Image.
 func WithID(id string) func(*Image) error {
 	return func(img *Image) error {
-		if img == nil {
-			return errors.New("no go on empty image")
-		}
```

Comment:
I think this was added to avoid a panic in case `FromMappedData` failed and returned a nil image. In this case, I think that this should panic, since the error was not checked. Also, this error is kind of weird.

Reply:
(ditto throughout)

Reply:
(I'm also super open to suggestions!)

Reply:
The error is weird but panicking too. Maybe improving the error message is better 🤔  what do you think? Making it a friendly panic 😆